### PR TITLE
en: Deploy a TiDB Cluster across Multiple Kubernetes clusters

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-tidb-operator/pull/974
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+
+- The TiDB components on each Kubernetes cluster can access the Pod IP of all TiDB components in and between clusters.
+- The TiDB components on each Kubernetes cluster can parse the Pod FQDN of all TiDB components in and between clusters.
+
+## Supported scenarios
+
+Currently supported scenarios:
+
+- Newly deployed a TiDB cluster across multiple Kubernetes clusters.
+- Deploy new clusters that enable this feature on other Kubernetes clusters and join the clusters that also enable this feature.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:22]:
```suggestion
- Deploy a new TiDB cluster that enables this feature on other Kubernetes clusters and join the cluster with another TiDB cluster that also enables this feature.
```

Comment (ran-huang) [reply en/deploy-tidb-cluster-across-multiple-kubernetes.md:22]: @handlerww Please check whether my suggestion is correct. Thanks!

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+
+Experimental supported scenarios:
+
+- For clusters with existing data that disable this feature, change to enable this feature. If you need to use it in a production environment, it is recommended to complete this requirement through data migration.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:26]:
```suggestion
- Enable this feature for a cluster that already has data. If you need to perform this action in a production environment, it is recommended to complete this requirement through data migration.
```
@handlerww PTAL

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+
+Unsupported scenarios:
+
+- Two interconnected existing data clusters. This scenario should be completed through data migration.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:30]:
```suggestion
- You cannot interconnect two clusters that already have data. You might perform this action through data migration.
```
@handlerww PTAL

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+
+## Deploy a cluster across multiple Kubernetes clusters
+
+If you deploy a TiDB cluster across multiple Kubernetes clusters, by default, you have already deployed Kubernetes clusters required for this scenario, and then perform the following deployment on this basis.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:34]:
```suggestion
Before you deploy a TiDB cluster across multiple Kubernetes clusters, you need to first deploy the Kubernetes clusters required for this operation. The following deployment assumes that you have completed Kubernetes deployment.
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+
+The following takes the deployment of two clusters as an example. Cluster one is the initial cluster. Create it according to the configuration given below. After cluster one is running normally, create cluster two according to the configuration given below. After creating and deploying clusters, two clusters run normally.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:36]: `cluster one` and `cluster two` may be confusing. How about `cluster #1` and `cluster #2`, or `cluster A` and `cluster B`.
Can you update the doc?

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,662 @@
...
+  tidb:
+    baseImage: pingcap/tidb
+    replicas: 1
+    service:
+      type: ClusterIP
+    config: {}
+EOF
+```
+
+## Deploy the TiDB cluster with TLS enabled between TiDB components across multiple Kubernetes clusters
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:150]:
```suggestion
## Deploy the TLS-enabled TiDB cluster across multiple Kubernetes clusters
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,662 @@
...
+
+You can follow the steps below to enable TLS between TiDB components for TiDB clusters deployed across multiple Kubernetes clusters.
+
+### Issue the root certificate
+
+#### Issue the root certificate using `cfssl`
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:156]:
```suggestion
#### Use `cfssl`
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,662 @@
...
+
+If you use `cfssl`, the CA certificate issue process is no different from the general issue process. You need to save the CA certificate created for the first time, and use this CA certificate when issuing certificates for TiDB components later. When creating a component certificate in a cluster, you do not need to create a CA certificate again and only need to complete step one to four in the [Enabling TLS between TiDB components](enable-tls-between-components.md#using-cfssl) once to complete the issue of the CA certificate. You need to start from step five for the issue of certificates between other cluster components.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:158]:
```suggestion
If you use `cfssl`, the CA certificate issue process is the same as the general issue process. You need to save the CA certificate created for the first time, and use this CA certificate when you issue certificates for TiDB components later. 

In other words, when you create a component certificate in a cluster, you do not need to create a CA certificate again. Complete step 1 ~ 4 in [Enabling TLS between TiDB components](enable-tls-between-components.md#using-cfssl) once to issue the CA certificate. After that, start from step 5 to issue certificates between other cluster components.
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 9]
```diff
@@ -0,0 +1,662 @@
...
+
+#### Use the `cert-manager` system to issue a root certificate
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:160]:
```suggestion
#### Use `cert-manager`
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,662 @@
...
+
+If you use `cert-manager`, you only need to create a `CA Issuer` and a `CA Certificate` in the initial cluster, and export the `CA Secret` to other new clusters that want to join. Other clusters only need to create component certificates to issue `Issuer` (Refers to the Issuer named ${cluster_name}-tidb-issuer in the [TLS document](enable-tls-between-components.md#using-cert-manager
+)). Use this CA to configure `Issuer`, the detailed process is as follows:
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:163]:
```suggestion
If you use `cert-manager`, you only need to create a `CA Issuer` and a `CA Certificate` in the initial cluster, and export the `CA Secret` to other new clusters that want to join. 

For other clusters, you only need to create a component certificate `Issuer` (refers to `${cluster_name}-tidb-issuer` in the [TLS document](enable-tls-between-components.md#using-cert-manager)) and configure the `Issuer` to use the `CA`. The detailed process is as follows:
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,662 @@
...
+      renewBefore: 720h # 30d
+      issuerRef:
+        name: ${cluster_name}-selfsigned-ca-issuer
+        kind: Issuer
+    EOF
+    ```
+
+2. Export the CA and delete irrelevant information.
+
+  First, you need to export the `Secret` that stores the CA. The name of the `Secret` can be obtained from the `.spec.secretName` of the first step `Certificate`.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:209]: Please indent 4 spaces

Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:209]:
```suggestion
  First, you need to export the `Secret` that stores the CA. The name of the `Secret` can be obtained from `.spec.secretName` of the `Certificate` YAML file in the first step.
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 12]
```diff
@@ -0,0 +1,662 @@
...
+
+  {{< copyable "shell-regular" >}}
+
+  ```bash
+  kubectl get secret cluster1-ca-secret -n ${namespace} -o yaml > ca.yaml
+  ```
+
+  Delete irrelevant information in the `Secret YAML` file. The `YAML` file after deletion is as follows, where the information in `data` has been omitted:
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:217]:
```suggestion
  Delete irrelevant information in the Secret YAML file. After the deletion, the YAML file is as follows (the information in `data` is omitted):
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 13]
```diff
@@ -0,0 +1,662 @@
...
+
+3. Import the exported CA to other clusters.
+
+   You need to configure the `namespace` so that related components can access the CA certificate:
+
+  {{< copyable "shell-regular" >}}
+
+    ```bash
+    kubectl apply -f ca.yaml -n ${namespace}
+   ·```
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:239]:
```suggestion
    ```
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 14]
```diff
@@ -0,0 +1,662 @@
...
+
+4. Create a component certificate in the initial cluster and the new cluster to issue `Issuer` using this CA.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:241]:
```suggestion
4. Create a component certificate `Issuer` in the initial cluster and the new cluster, and configure it to use this CA.
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 15]
```diff
@@ -0,0 +1,662 @@
...
+
+    1. Create a certificate issuing `Issuer` between TiDB components in the initial cluster.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:243]:
```suggestion
    1. Create an `Issuer` that issues certificates between TiDB components in the initial cluster.
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 16]
```diff
@@ -0,0 +1,662 @@
...
+        metadata:
+          name: ${cluster_name}-tidb-issuer
+          namespace: ${namespace}
+        spec:
+          ca:
+            secretName: ${ca_secret_name}
+        EOF
+        ```
+
+    2. Create a certificate issuing `Issuer` between TiDB components in the new cluster.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:272]: ditto

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 17]
```diff
@@ -0,0 +1,662 @@
...
+
+       Set the following environment variables according to the actual situation. Among them, `ca_secret_name` needs to point to the `Secret` that you just import to store the `CA`. You can use the `cluster_name` and `namespace` in the following operations:
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:274]:
```suggestion
       Set the following environment variables according to the actual situation. Among them, `ca_secret_name` points to the imported `Secret` that stores the `CA`. You can use the `cluster_name` and `namespace` in the following operations:
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 18]
```diff
@@ -0,0 +1,662 @@
...
+       EOF
+       ```
+
+### Issue certificates for the TiDB components of each Kubernetes cluster
+
+You need to issue a component certificate for each TiDB component on the Kubernetes cluster. When issuing a component certificate, you need to add an authorization record ending with `.${cluster_domain}` to the hosts, for example, `${cluster_name}-pd.${namespace}.svc.${cluster_domain}`.
+
+#### Use the cfssl system to issue certificates for TiDB components
+
+If you use `cfssl`, take the certificate used to create the PD component as an example, the `pd-server.json` file is as follows.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:306]:
```suggestion
The following example shows how to use `cfssl` to create a certificate used by PD. The `pd-server.json` file is as follows.
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 19]
```diff
@@ -0,0 +1,662 @@
...
+
+Set the following environment variables according to the actual situation.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:308]:
```suggestion
Set the following environment variables according to the actual situation:
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 20]
```diff
@@ -0,0 +1,662 @@
...
+            "ST": "San Francisco"
+        }
+    ]
+}
+EOF
+```
+
+#### Use the `cert-manager` system to issue certificates for TiDB components
+
+If you use `cert-manager`, take the certificate used to create the PD component as an example, `Certifcates` is shown below.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:359]:
```suggestion
The following example shows how to use `cert-manager` to create a certificate used by PD.  `Certifcates` is shown below.
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 21]
```diff
@@ -0,0 +1,662 @@
...
+You need to refer to the TLS-related documents, issue the corresponding certificates for the components, and create the `Secret` in the corresponding Kubernetes clusters.
+
+For other TLS-related information, refer to the following documents:
+
+- [Enable TLS between TiDB Components](enable-tls-between-components.md)
+- [Enable TLS for the MySQL Client](enable-tls-for-mysql-client.md)
+
+### Deploy the initial cluster
+
+To deploy and initialize the cluster, use the following command. In actual use, you need to set the contents of the `cluster1_name` and `cluster1_cluster_domain` variables according to your actual situation, where `cluster1_name` is the cluster name of cluster #1, `cluster1_cluster_domain` is the `Cluster Domain` of cluster #1, and `cluster1_namespace` is the namespace of cluster #1. The following `YAML` file enables the TLS feature, and each component starts to verify the certificates issued by the `CN` for the `CA` of `TiDB` by configuring the `cert-allowed-cn`.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:424]:
```suggestion
This section introduces how to deploy and initialize the cluster. 

In actual use, you need to set the contents of the `cluster1_name` and `cluster1_cluster_domain` variables according to your actual situation, where `cluster1_name` is the cluster name of cluster #1, `cluster1_cluster_domain` is the `Cluster Domain` of cluster #1, and `cluster1_namespace` is the namespace of cluster #1. The following `YAML` file enables the TLS feature, and each component starts to verify the certificates issued by the `CN` for the `CA` of `TiDB` by configuring the `cert-allowed-cn`.
```

**Review by ran-huang — COMMENTED** [review]
L425 and above are reviewed.

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 22]
```diff
@@ -0,0 +1,668 @@
...
+
+Take the cluster #2 created in the above document as an example. First, set the number of copies of PD, TiKV, TiDB to `0`. If you enable other components such as TiFlash, TiCDC, Pump, etc., set the number of these copies to `0`:
+
+{{< copyable "shell-regular" >}}
+
+```bash
+kubectl patch tc cluster2 --type merge -p '{"spec":{"pd":{"replicas":0},"tikv":{"replicas":0},"tidb":{"replicas":0}}}'
+```
+
+Wait for the status of cluster #2 to become `Ready`, and scale in related components to `0` copy:
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:577]:
```suggestion
Wait for the status of cluster #2 to become `Ready`, and scale in related components to `0` replica:
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 23]
```diff
@@ -0,0 +1,668 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```bash
+kubectl get pods -l app.kubernetes.io/instance=cluster2 -n pingcap
+```
+
+The Pod list is displayed as `No resources found.`. At this time, Pods have all been scaled in, and cluster #2 exits the cluster. Check the cluster status of cluster #2:
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:585]:
```suggestion
The Pod list shows `No resources found`. At this time, Pods have all been scaled in, and cluster #2 exits the cluster. Check the cluster status of cluster #2:
```

**Review by ran-huang — COMMENTED** [review]
L600 and above are reviewed

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 24]
```diff
@@ -0,0 +1,670 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```bash
+kubectl delete tc cluster2
+```
+
+Through the above steps, you can complete exit and resources reclaim of the joined clusters.
+
+## Enable the existing data cluster across multiple Kubernetes cluster feature as the initial TiDB cluster
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:605]:
```suggestion
## Enable the feature for a cluster with existing data and make it the initial TiDB cluster
```

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 25]
```diff
@@ -0,0 +1,670 @@
...
+
+    ```bash
+    kubectl patch tidbcluster cluster1 --type merge -p '{"spec":{"clusterDomain":"cluster1.com"}}'
+    ```
+
+    After completing the modification, the TiDB cluster performs the rolling update.
+
+2. Update the `PeerURL` information of PD:
+
+    After completing the rolling update, you need to use `port-forward` to expose PD's API interface, and use API interface of PD to update `PeerURL` of PD.
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:629]:
```suggestion
    After completing the rolling update, you need to use `port-forward` to expose PD's API and use the API to update `PeerURL` of PD.
```

Comment (ran-huang) [reply en/deploy-tidb-cluster-across-multiple-kubernetes.md:629]: API is short for application programming interface. There's no need to say `API interface`. Please update other expressions in this doc as well.

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md [hunk 26]
```diff
@@ -0,0 +1,670 @@
...
+
+    1. Use `port-forward` to expose API interface of PD:
+
+        {{< copyable "shell-regular" >}}
+
+        ```bash
+        kubectl port-forward pods/cluster1-pd-0 2380:2380 2379:2379 -n pingcap
+        ```
+
+    2. Access `PD API` to obtain `members` information. Note that after using `port-forward`, the terminal is occupied. You need to perform the following operations in another terminal:
```
Comment (ran-huang) [inline en/deploy-tidb-cluster-across-multiple-kubernetes.md:639]:
```suggestion
    2. Access `PD API` to obtain `members` information. Note that after using `port-forward`, the terminal session is occupied. You need to perform the following operations in another terminal session:
```
@handlerww Is this what you mean?

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md (hunk 1 above)
Comment (handlerww) [reply en/deploy-tidb-cluster-across-multiple-kubernetes.md:22]: Actually, we can deploy many clusters to join the initial cluster.  Should we cover this knowledge?

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md (hunk 26 above)
Comment (handlerww) [reply en/deploy-tidb-cluster-across-multiple-kubernetes.md:639]: Yes, it means that the Linux shell terminal session will be occupied.

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md (hunk 24 above)
Comment (handlerww) [reply en/deploy-tidb-cluster-across-multiple-kubernetes.md:605]: The suggestion is better.

Comment (handlerww) [reply en/deploy-tidb-cluster-across-multiple-kubernetes.md:605]: The suggestion is more accurate.

File: en/deploy-tidb-cluster-across-multiple-kubernetes.md (hunk 1 above)
Comment (ran-huang) [reply en/deploy-tidb-cluster-across-multiple-kubernetes.md:22]: Sure. How about this?

```suggestion
- Deploy new TiDB clusters that enable this feature on other Kubernetes clusters and join the initial TiDB cluster.
```

**Review by ran-huang — APPROVED** [review]
LGTM.
@handlerww PTAL~

**Comment by ran-huang** [comment]
@Joyinqin Please address the comments and apply the changes to the Chinese doc if applicable. Thanks!

**Comment by Joyinqin** [comment]
@handlerww PTAL again~thx

**Review by handlerww — APPROVED** [review]

**Review by handlerww — APPROVED** [review]
LGTM

**Pull request merged**